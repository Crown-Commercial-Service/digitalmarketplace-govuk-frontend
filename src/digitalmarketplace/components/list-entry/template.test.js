/**
 * @jest-environment jsdom
 */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('list-entry')

describe('List entry', () => {
  describe('Describe when no JavaScript', () => {
    it('default example passes accessibility tests', async () => {
      const $ = render('list-entry', examples.default)

      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })
    describe('fieldset', () => {
      it('renders the component with a legend', () => {
        const $ = render('list-entry', examples.default)
        const legend = $('.dm-list-entry__legend').text().trim()
        expect(legend).toBe(examples.default.fieldset.legend.text)
      })
    })

    describe('hint text', () => {
      it('renders the component with hint text when it has been provided', () => {
        const $ = render('list-entry', examples.default)
        const hint = $('.govuk-hint').text().trim()
        expect(hint).toBe(examples.default.hint.text)
      })
    })

    describe('list of entries', () => {
      it('renders the component with the correct number of items it was given', () => {
        const $ = render('list-entry', examples.default)
        expect($('.govuk-input').length).toBe(examples.default.maxItems)
      })
    })
  })
})
