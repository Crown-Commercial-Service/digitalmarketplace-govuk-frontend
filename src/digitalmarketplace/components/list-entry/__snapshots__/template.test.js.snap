// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List entry Default with no JavaScript renders a header component with all our standard links 1`] = `
"<html><head></head><body><div class=\\"app-list-entry\\">
<fieldset class=\\"govuk-fieldset\\">
  <legend class=\\"govuk-fieldset__legend app-list-entry__legend\\">
    What requirements are essential to carry out the work?
  </legend>

    <span id=\\"undefined-hint\\" class=\\"govuk-hint\\">
    Include the technical features of your product, eg graphical workflow, remote access. (Maximum 10 words per feature. Maximum 10 features.)
  </span>

    
    <input class=\\"govuk-input\\" id name type=\\"text\\" aria-describedby=\\"undefined-hint\\">

    
    <input class=\\"govuk-input\\" id name type=\\"text\\" aria-describedby=\\"undefined-hint\\">

    
    <input class=\\"govuk-input\\" id name type=\\"text\\" aria-describedby=\\"undefined-hint\\">

    
    <input class=\\"govuk-input\\" id name type=\\"text\\" aria-describedby=\\"undefined-hint\\">

</fieldset>
</div>--
</body></html>"
`;
